install_script.sh.template

```diff
@@ -1234,6 +1259,44 @@ if [ -n "$docker_injection_enabled" ]; then
 fi
 
 # Creating or overriding the install information
+function generate_install_id() {
+  # Try generating a UUID based on /proc/sys/kernel/random/uuid
+  uuid=$(cat /proc/sys/kernel/random/uuid 2>/dev/null)
+  # If that does not work, then try uuidgen
+  if [ ${#uuid} -ne 36 ]; then
+    uuid=$(uuidgen 2>/dev/null)
+  fi
+  # Convert to lowercase
+  uuid=$(echo "$uuid" | tr '[:upper:]' '[:lower:]')
+  printf "$uuid"
+}
+
+function generate_install_signature() {
+  local install_id="$1"
+  local install_type="$2"
+  local install_time="$3"
+  printf "{\"install_id\":\"$install_id\",\"install_type\":\"$install_type\",\"install_time\":$install_time}"
+}
+
+install_id=$(generate_install_id)
+install_type="linux_agent_installer"
+install_time=$(date +%s)
+
+# If an install.json already exists, is formatted correctly, and was generated by this script,
+# reuse the original install ID and time.
+if [ -f "$etcdir/install.json" ]; then
+  # Parse the JSON file using substring extraction to avoid a dependency on any JSON parser
+  install_info=$(cat "$etcdir/install.json" 2>/dev/null)
+  if [ ${#install_info} -eq 118 ]; then
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ If we keep the existing `install_info` and new `install.json` files separate, the naming here is a little confusing. At first read I thought it was reading the existing file.